esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5987,15 +5985,12 @@ StringBuffer &CWsDfuEx::getFileDafilesrvKeyName(StringBuffer &keyPairName, IDist
     return keyPairName;
 }
 
-void CWsDfuEx::getFileAccess(IEspContext &context, IUserDescriptor *udesc, SecAccessFlags accessType, IEspDFUFileAccessRequest &req, IEspDFUFileAccessResponse &resp)
+void CWsDfuEx::getFileAccess(IEspContext &context, IUserDescriptor *udesc, SecAccessFlags accessType, const char *logicalFileName, const char *jobId, IConstDFUFileAccessRequestBase &req, IEspDFUFileAccessInfo &accessInfo)
 {
     bool writePermissions = (accessType == SecAccess_Write) || (accessType == SecAccess_Full);
     bool readPermissions = true; // by implication
 
-    StringBuffer fileName(req.getName());
-    if (!isEmptyString(req.getCluster()))
-        fileName.append("@").append(req.getCluster());
-
+    VStringBuffer fileName("%s@%s", logicalFileName, req.getCluster());
```

Comment:
I think this will now be wrong if there is getCluster is blank (which it's allowed to be as optional)

Reply:
no